docs/design/kademlia_audit_gating.md

```diff
@@ -0,0 +1,94 @@
+# Kademlia Audit Gating
+
+## Abstract
+
+StorageNode B is added to StorageNode A's routing table only if StorageNode A trusts Satellite C and Satellite C as verified StorageNode B has a high enough identity-generate difficulty (CPU proof-of-work) and disk space. If StorageNode B does not have this verification, it is added to StorageNode A's Routing Table Antechamber.
+
+## Background
+
+### White paper: 4.6.1 Mitigating Sybil attacks
+While we’ve adopted the proof-of-work scheme S/Kademlia proposes to partially address Sybil attacks, we extend Kademlia with an application specific integration to further defend our network. 
+
+
+Given two storage nodes, A and B, storage node B is not allowed to enter storage node A’s routing table until storage node B can present a signed message from a Satellite C that storage node A trusts claiming that B has passed enough audits that C trusts it (sections 4.13 and 4.15). This ensures that only nodes with verified disk space have the opportunity to participate in the routing layer.
+
+
+A node that is allowed to enter routing tables is considered vetted and lookups only progress through vetted nodes. To make sure unvetted nodes can still be found, vetted nodes keep unbounded lists of their unvetted neighbors provided that the XOR distance to all unvetted neighbors is no farther than the farthest of the k-closest vetted neighbors. Unvetted nodes keep their k-nearest vetted nodes up-to-date.
+
+
+## Goals
+1. Add a signed message from Satellites to authenticate whether a Node has a high enough ID generation difficulty and disk space.
+
+2. Create a trusted Satellite list that contains the IDs of Satellites from which Nodes will accept verification signatures of other Nodes.
+
+3. Implement a Routing Table Antechamber: An XOR-ordered data structure in which unverified nodes are entered if successfully contacted. 
+
+4. Modify FindNear to return n XOR-closest nodes from the antechamber in addition to those already returned from the routing table.
+
+5. On deployment, avoid complete erasure of the network's routing tables.
+
+
+## Terminology
+**Routing Table Antechamber** - *XOR-ordered temporary holding place for unverified storagenodes*
+
+**Node Neighborhood** - *The k-closest nodes to self where distance is measured by XOR. A node is within the node neighborhood if it is closer than the furthest node in the neighborhood. The vetted node neighborhood is the k-closest nodes that are currently in the Routing Table.*
+
+[TODO Node Neighborhood Diagram]
+
+## Design
+
+1. Satellite Signatures for Node Verification
+    - Identities can sign messages already
+    - Create a Message (protobuf) to sign that Satellite C says Node B has been vetted
```

Comment:
I think a draft of the message struct could be good to see here so we can decide fields and potential methods in advance.

Reply:
good idea